OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,457 @@
+            patcher.AddPostfix<DoubleAxisCommand>("Update", typeof(InputInterceptor), nameof(InputInterceptor.DoubleAxisUpdatePost));
+            Instance = this;
+        }
+
+        private void UpdateCombo()
+        {
+            if (Time.realtimeSinceStartup - _lastUpdate > _updateCooldown)
+            {
+                _lastUpdate = Time.realtimeSinceStartup;
+                Int64 hash = 0;
+                int[] keys = new int[_maxComboLength];
+                int keysCount = 0;
+                bool countdownTrigger = false;
+                for (int code = _minUsefulKey; code < _maxUsefulKey; code++)
+                {
+                    if (Enum.IsDefined(typeof(KeyCode), (KeyCode)code) && UnityEngine.Input.GetKey((KeyCode)code))
+                    {
+                        keys[keysCount] = code;
+                        keysCount++;
+                        if (keysCount > _maxComboLength)
+                        {
+                            hash = (int)RegistrationCode.CombinationTooLong;
+                            break;
+                        }
+                        hash = hash * _maxUsefulKey + code;
+                        if (Time.realtimeSinceStartup - _timeout[code] < _cooldown)
+                            countdownTrigger = true;
+                    }
+                }
+                if (_comboRegistry.ContainsKey(hash))
+                {
+                    IModInputCombination combination = _comboRegistry[hash];
+                    if (combination == _lastPressed || !countdownTrigger)
+                    {
+                        if (_lastPressed != null)
+                            _lastPressed.SetPressed(false);
+                        _lastPressed = (ModInputCombination)combination;
+                        _lastPressed.SetPressed();
+                        for (int i = 0; i < keysCount; i++)
+                            _timeout[keys[i]] = Time.realtimeSinceStartup;
+                        return;
+                    }
+                }
+                if (_lastPressed != null)
+                {
+                    _lastPressed.SetPressed(false);
+                }
+                _lastPressed = null;
+            }
+        }
+
+        private bool IsPressed_Combo(IModInputCombination combination)
+        {
+            UpdateCombo();
+            return _lastPressed == combination;
+        }
+
+        private bool IsNewlyPressed_Combo(IModInputCombination combination, bool keep = false)
+        {
+            UpdateCombo();
+            return _lastPressed == combination && combination.IsFirst(keep);
+        }
+
+        private bool WasTapped_Combo(IModInputCombination combination)
+        {
+            UpdateCombo();
+            return combination != _lastPressed && (Time.realtimeSinceStartup - combination.GetLastPressedMoment() < _tapKeep) && (combination.GetPressDuration() < _tapDuration);
```

Comment:
Seems like you do `Time.realtimeSinceStartup - combination.GetLastPressedMoment() < _tapKeep` a bunch of times in this file. So this whole expression should have its own method. I guess either in ModInputHandler.cs, or in ModInputCombination.cs (passing `_tapKeep` as an argument).

Reply:
didn't make things much shorter, but probably easier to read (also made it multiline)